foia_hub/templates/contacts/index.html

```diff
@@ -13,26 +13,34 @@
     {% endif %}
     {% if not query %}
       <h2><em>Federal Agencies</em></h2>
-    {% endif %}    
-  
-    <ul>
-      {% for agency in agencies %}
-          <li>
-            <div class="agencies--name">
-              <h3><a href="/contacts/{{agency.slug }}">{{ agency.name }}</a></h3>
-            </div>
-            <div class="agencies--description">
-              <p>
-                {% if agency.description %}
-                  {{ agency.description }}
-                {% else %}
-                  Click to find out more about this agency.
-                {% endif %}
-              </p>
-            </div>
-          </li>
+    {% endif %}
+      <h3>Results: {{agencies|length}}</h3>
+      <p>
+          {% for letter in agencies|groupby('name.0') %}
```

Comment:
Looks like:
```
A C R S

A
AStar Agency

C
Candidate Agency

R
Red Agency

S
Super Agency
```

The listing at the top is a separate loop from the listing at the bottom. Hypothetically, you could combine them into a single loop using concatenated templates but that sounds very ugly.